front/lib/api/assistant/conversation.ts

```diff
@@ -455,34 +553,16 @@ export async function getConversation(
     ],
   });
 
-  const render = await Promise.all(
-    messages.map((message) => {
-      return (async () => {
-        if (message.userMessage) {
-          const m = await renderUserMessage(message, message.userMessage);
-          return { m, rank: message.rank, version: message.version };
-        }
-        if (message.agentMessage) {
-          const m = await renderAgentMessage(auth, {
-            message,
-            agentMessage: message.agentMessage,
-            messages,
-          });
-          return { m, rank: message.rank, version: message.version };
-        }
-        if (message.contentFragment) {
-          const m = await renderContentFragment({
-            message: message,
-            contentFragment: message.contentFragment,
-          });
-          return { m, rank: message.rank, version: message.version };
-        }
-        throw new Error(
-          "Unreachable: message must be either user, agent or content fragment"
-        );
-      })();
-    })
-  );
+  const [userMessages, agentMessages, contentFragments] = await Promise.all([
+    batchRenderUserMessages(messages.filter((m) => !!m.userMessage)),
```

Comment:
should we always send all messages and defer filtering to the function itself. Will make that code more easy to understand otherwise it's very unclear why there are special cases here and may look like a bug?